### PR TITLE
[FLINK-25537][JUnit5 Migration] Module: flink-core with,Package: types

### PR DESCRIPTION
**Pull request by GOODBOY008**
Changes:

- Migrate Module: flink-core with,Package: types to junit5

### CODE REVIEW

**Comment by GOODBOY008** [comment]
@Jiabao-Sun @1996fanrui PTAL

File: flink-core/src/test/java/org/apache/flink/types/StringSerializationTest.java [hunk 1]
```diff
@@ -33,226 +33,166 @@
 import java.util.function.BiConsumer;
 import java.util.function.Function;
 
-import static org.junit.Assert.assertArrayEquals;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
+import static org.assertj.core.api.Assertions.assertThat;
 
 /** Test for the serialization of Strings through the StringValue class. */
 public class StringSerializationTest {
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/types/StringSerializationTest.java:39]:
```suggestion
class StringSerializationTest {
```

File: flink-core/src/test/java/org/apache/flink/types/StringValueSerializationTest.java [hunk 2]
```diff
@@ -22,96 +22,71 @@
...
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.util.Random;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
+import static org.assertj.core.api.Assertions.assertThat;
 
 /** Test for the serialization of StringValue. */
 public class StringValueSerializationTest {
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/types/StringValueSerializationTest.java:35]:
```suggestion
class StringValueSerializationTest {
```

File: flink-core/src/test/java/org/apache/flink/types/PrimitiveDataTypeTest.java [hunk 3]
```diff
@@ -134,106 +129,82 @@ public void testStringValue() {
...
-        Assert.assertTrue(string0.compareTo(string6) == 0);
-        Assert.assertTrue(stringIsA.compareTo(string7) == 0);
+        assertThat(string0.compareTo(string0)).isZero();
+        assertThat(string0.compareTo(string1)).isZero();
+        assertThat(string0.compareTo(string2)).isPositive();
+        assertThat(string0.compareTo(string3)).isNegative();
+        assertThat(chars5).isEqualTo(stringThis);
+        assertThat(stringThis.compareTo(string5)).isZero();
+        assertThat(string0.compareTo(string6)).isZero();
+        assertThat(stringIsA.compareTo(string7)).isZero();
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/types/PrimitiveDataTypeTest.java:139]: assertThatComparable(string0).isEqualByComparingTo(string0);

We can use `assertThatComparable` here.

File: flink-core/src/test/java/org/apache/flink/types/PrimitiveDataTypeTest.java [hunk 4]
```diff
@@ -134,106 +129,82 @@ public void testStringValue() {
...
-        try {
-            string7.charAt(5);
-            Assert.fail(
-                    "Exception should have been thrown when accessing characters out of bounds.");
-        } catch (IndexOutOfBoundsException iOOBE) {
-            // expected
-        }
+        assertThat(stringIsA.compareTo(string7)).isZero();
+        assertThat(string0.compareTo(string7)).isNegative();
+        assertThat(string7.hashCode()).isEqualTo(stringIsA.hashCode());
```
Comment (1996fanrui) [inline flink-core/src/test/java/org/apache/flink/types/PrimitiveDataTypeTest.java:146]: assertThatComparable and hasSameHashCode

**Review by 1996fanrui — COMMENTED** [review]
Thanks @GOODBOY008 for the contribution!

Overall is fine for me, I only left some minor comments, please take a look in your free time, thanks~

**Review by 1996fanrui — APPROVED** [review]
Thanks for the quick update!

LGTM

**Comment by 1996fanrui** [comment]
Hi @GOODBOY008 , would you mind rebasing master branch before merging? It can ensure the master branch works well after merging.

Also, I will merge this PR if no other comments within 2 days.

cc @Jiabao-Sun

**Review by Jiabao-Sun — APPROVED** [review]
Thanks @GOODBOY008 for this great work and @1996fanrui for the hard review.
LGTM.

**Pull request merged**